example/glue/EnvironmentDelegate.cpp

```diff
@@ -21,9 +21,40 @@
 #include "omrExampleVM.hpp"
 
 #include "AtomicOperations.hpp"
-#include "EnvironmentBase.hpp"
+#include "EnvironmentStandard.hpp"
 #include "EnvironmentDelegate.hpp"
 #include "GCExtensionsBase.hpp"
+#include "SublistFragment.hpp"
+
+OMR_VMThread *
+MM_EnvironmentDelegate::attachVMThread(OMR_VM *omrVM, const char *threadName, uintptr_t reason)
+{
+	OMR_VMThread *omrVMThread = NULL;
+	omr_error_t rc = OMR_ERROR_NONE;
+
+	rc = OMR_Glue_BindCurrentThread(omrVM, threadName, &omrVMThread);
+	if (OMR_ERROR_NONE != rc) {
+		return NULL;
+	}
+	return omrVMThread;
+}
+
+void
+MM_EnvironmentDelegate::detachVMThread(OMR_VM *omrVM, OMR_VMThread *omrVMThread, uintptr_t reason)
+{
+	if (NULL != omrVMThread) {
+		OMR_Glue_UnbindCurrentThread(omrVMThread);
+	}
+}
+
+void
+MM_EnvironmentDelegate::flushNonAllocationCaches()
+{
+#if defined(OMR_GC_MODRON_SCAVENGER)
+	MM_EnvironmentStandard *envStd = MM_EnvironmentStandard::getEnvironment(_env);
```

Comment:
I agree with you that languages may have caches they need to flush and this is the right API I do not believe the scavenger rs caches should be a language requirement.  It should be part of OMR